tutorials/05-linear-regression/05_linear-regression.jmd

```diff
@@ -153,16 +154,14 @@ using GLM
 train_with_intercept = hcat(ones(size(train, 1)), train)
 ols = lm(train_with_intercept, train_target)
 
-# Compute predictions on the training data set
-# and unstandardize them.
-p = GLM.predict(ols)
-train_prediction_ols = μtarget .+ σtarget .* p
+# Compute predictions on the training data set and unstandardize them.
+train_prediction_ols = GLM.predict(ols)
+StatsBase.reconstruct!(dt_targets, train_prediction_ols)
```

Comment:
so the inner working of `StatsBase.reconstruct` is: it first calculates the mean and std from the training targets, and then apply them to the predicted (scaled) targets?

Reply:
No. You compute a transformation object with `fit` which in the case of `ZScoreTransform` mainly stores empirical mean and standard deviation/variance of each feature. Then you can apply this transformation with `transform`/`transform!` which in this case will normalize the data. And with `reconstruct`/`reconstruct!` you can un-normalize data by scaling it with the learnt standard deviation and shifting it with the mean.